content/blog/exposing-an-application-using-ingress-and-tls-termination-on-kubernetes-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,528 @@
+├── ingress-path-based-selfsigned.yaml
+└── README.md
+```
+
+Each YAML manifest file in the folder *'apps'* defines the *Deployment* and the *Service* resource.  
+
+T﻿ype the following commands to deploy those Nginx applications to the namespace *nginx-apps*:
+
+```shell
+$ cd ingress-demo/
+$ kubectl apply -f apps/nginx-main.yaml -n nginx-apps
+service/nginx-main created
+deployment.apps/nginx-main created
+$ kubectl apply -f apps/nginx-green.yaml -n nginx-apps
+service/nginx-green created
+deployment.apps/nginx-green created
+$ kubectl apply -f apps/nginx-blue.yaml -n nginx-apps
+service/nginx-blue created
+deployment.apps/nginx-blue created
+```
+
+Type the command shown below to check the details of application deployment:  
+
+```shell
+$ kubectl get all -n nginx-apps
+NAME                              READY   STATUS    RESTARTS   AGE
+pod/nginx-blue-78647f4c4b-z8wq9   1/1     Running   0          10s
+pod/nginx-green-8956bbd9f-zz7hk   1/1     Running   0          22s
+pod/nginx-main-64bfd77895-tf7xd   1/1     Running   0          31s
+
+NAME                  TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
+service/nginx-blue    ClusterIP   10.108.51.116   <none>        80/TCP    15s
+service/nginx-green   ClusterIP   10.106.115.65   <none>        80/TCP    23s
+service/nginx-main    ClusterIP   10.108.33.44    <none>        80/TCP    32s
+
+NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/nginx-blue    1/1     1            1           15s
+deployment.apps/nginx-green   1/1     1            1           24s
+deployment.apps/nginx-main    1/1     1            1           32s
+
+NAME                                    DESIRED   CURRENT   READY   AGE
+replicaset.apps/nginx-blue-78647f4c4b   1         1         1       15s
+replicaset.apps/nginx-green-8956bbd9f   1         1         1       24s
+replicaset.apps/nginx-main-64bfd77895   1         1         1       32s
+```
+
+Three Nginx services, *nginx-main*, *nginx-blue* and *nginx-green*, are deployed as the *ClusterIP* type. They provide internal connectivity and can solely be accessed from within the cluster.
+
+T﻿ype the following commend to check that all the application service endpoints have been populated:
+
+```shell
+$ kubectl get endpoints -n nginx-apps
+NAME          ENDPOINTS        AGE
+nginx-blue    10.192.3.78:80    1m
+nginx-green   10.192.4.45:80    1m
+nginx-main    10.192.4.44:80    1m
+```
+
+### Set up Ingress TLS
+
+The Ingress resource with TLS has to be created. Here is the sample Ingress TLS resource *ingress-host-based-selfsigned.yaml*, available from the GitHub repo [ingress-demo](https://github.com/GuopingJia/ingress-demo):
+
+```shell
+$ cat ingress-host-based-selfsigned.yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-host-based-selfsigned
+  annotations:
+    ingress.kubernetes.io/ssl-redirect: "true"
+    cert-manager.io/issuer: "nginx-selfsinged-issuer"
```

Comment:
is there a typo here for the cert-manager.io/issuer ??
Shouldn't it be: ngnix-selfsigned-issuer rather than ngnix-selfsinged-issuer ???

Reply:
Good catch ! fixed it (also the code in the github repo).